dali/sasha/saxref.cpp

```diff
@@ -911,17 +951,43 @@ class CNewXRefManager: public CNewXRefManagerBase
                 fname.toLowerCase();
             addPathSepChar(path).append(fname);
             if (iter->isDir())  {
-                if (!dirFiltered(path.str())) {
-                    dirs.append(fname.str());
+                // Check if subdirectory is a dirPerPart or stripe directory
+                const char *dir = fname.str();
+                bool isStriped = dir[0] == 'd';
+                if (isStriped)
+                    dir++;
+                bool isSpecialDir = true;
+                while (isSpecialDir && *dir) {
+                    if (!isdigit(*(dir++)))
+                        isSpecialDir = false;
```

Comment:
shouldn't this set isStriped to false too?

Reply:
also, as it stands if just 'd' , isStriped will remain true when it should not (it would not be a isSpecialDir either).

Reply:
> also, as it stands if just 'd' , isStriped will remain true when it should not (it would not be a isSpecialDir either).

I think this is still true.. as the code stands, it won't enter the wild loop because if just "d", *dir will be \0. So will leave isDirStriped and isSpecialDir set to true incorrectly.